### PR TITLE
Fix the RLP serialization of Stint values

### PR DESCRIPTION
**Pull request by zah**
To understand the code, you must know that the RLP spec has some
peculiarities when encoding integers:

1) The zero value must be encoded as an empty list
2) Values below 128 are encoded as a single byte
3) Vaues above 128 must be encoded with the minimum number of bytes
   possible.

### CODE REVIEW

File: eth_common/eth_types.nim [hunk 1]
```diff
@@ -121,14 +121,33 @@ else:
 # Rlp serialization:
 #
 
+func trailingNonZeros(val: openarray[byte]): int =
+  for i in countdown(val.len - 1, 0):
+    if val[i] == 0:
+      return val.len - i - 1
+
+  return val.len
```
Comment (mratsim) [inline eth_common/eth_types.nim]: You will have a corner case for negative int, for example for a uint16:

10000000 11111111 --> should we take that as 1 trailing non zero, ignoring the sign or 2?

Note that there is a "countLeadingZeroBits" in Stint for uint but I don't think it maps cleanly to bytes (https://github.com/status-im/nim-stint/blob/master/stint/uint_public.nim)

File: eth_common/eth_types.nim [hunk 2]
```diff
@@ -121,14 +121,33 @@ else:
...
+
 proc read*(rlp: var Rlp, T: typedesc[Stint|StUint]): T {.inline.} =
   if rlp.isBlob:
-    result = readUintBE[result.bits](rlp.toBytes.toOpenArray)
+    let bytes = rlp.toBytes
+    if bytes.len > 0:
+      result = readUintBE[result.bits](bytes.toOpenArray)
+    else:
```
Comment (mratsim) [inline eth_common/eth_types.nim]: Shouldn't there be a special case for value < 128? I'd like a test to make sure serialization/deserializing works properly.

Comment (zah) [reply eth_common/eth_types.nim]: The special case is covered by the `isBlob` check. See the use of `rlp.getByteValue.to(T)` below.

File: eth_common/eth_types.nim (hunk 1 above)
Comment (zah) [reply eth_common/eth_types.nim]: Good catch. The RLP spec is defined only over unsigned integers, so I'll change the signature of the procs to handle only UInt values.

**Comment by zah** [comment]
Review comments addressed. A separate issue was created for adding the tests later:
https://github.com/status-im/nim-eth-common/issues/5

**Pull request merged**